src/main.rs

```diff
@@ -283,7 +283,7 @@ fn main() {
         let working_dir = tmp_path.join(format!("{}", i));
         let source_root = source_root.clone();
 
-        let t = thread::spawn(move || {
+        let t = thread::Builder::new().name(format!("Consumer {}",&i.to_string())).spawn(move || {
```

Comment:
You probably don't need to turn `i` into a string, format! should be able to handle integers directly

Reply:
Done! Compiled without turning into string, indeed.